pkg/controller/appdefinition/deploy.go

```diff
@@ -32,6 +32,22 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func RemoveLabelsAndAnnotationsConfig(h router.Handler) router.Handler {
+	return router.HandlerFunc(func(req router.Request, resp router.Response) error {
+		appInstance := req.Object.(*v1.AppInstance)
+		cfg, err := config.Get(req.Ctx, req.Client)
+		if err != nil {
+			return err
+		}
+
+		if cfg.IgnoreUserLabelsAndAnnotations != nil && *cfg.IgnoreUserLabelsAndAnnotations {
```

Comment:
A tip @ibuildthecloud gave me: For nil bools feel free to add GetIgnoreUserLabelsAndAnnotations() method like the GetDevMode() method on appInstanceSpec. Makes the code more readable.

Reply:
I took a slightly different approach here because `config.Get` "completes" the config object with defaults.

Reply:
for my benefit..this is just to check for bools in AppInstanceSpec are there and set?

Reply:
Which part @jsilverio22 ? Really anywhere we have nilable bools, you can add a helper to do this type of logic, EXCEPT for config because as donnie is pointing out, it isnt necessary because the config.Get is responsible for ensure its not nil

Reply:
I was looking at the GetDevMode(). But that makes sense bundling it up for clarity and why it is not necessary in this case.